tests/mocha/shortcut_registry_test.js

```diff
@@ -0,0 +1,352 @@
+    });
+    test('Unregistering a shortcut with key mappings', function() {
+      var testShortcut = {'name': 'test_shortcut'};
+      this.registry.keyMap_['keyCode'] = ['test_shortcut'];
+      this.registry.registry_['test_shortcut'] = testShortcut;
+
+      this.registry.unregister('test_shortcut');
+
+      var shortcut = this.registry.registry_['test'];
+      var keyMappings = this.registry.keyMap_['keyCode'];
+      chai.assert.isUndefined(shortcut);
+      chai.assert.isUndefined(keyMappings);
+    });
+    test('Unregistering a shortcut with colliding key mappings', function() {
+      var testShortcut = {'name': 'test_shortcut'};
+      this.registry.keyMap_['keyCode'] = ['test_shortcut', 'other_shortcutt'];
+      this.registry.registry_['test_shortcut'] = testShortcut;
+
+      this.registry.unregister('test_shortcut');
+
+      var shortcut = this.registry.registry_['test'];
+      var keyMappings = this.registry.keyMap_['keyCode'];
+      chai.assert.lengthOf(keyMappings, 1);
+      chai.assert.isUndefined(shortcut);
+    });
+  });
+
+  suite('addKeyMapping', function() {
+    test('Adds a key mapping', function() {
+      this.registry.registry_['test_shortcut'] = {'name': 'test_shortcut'};
+
+      this.registry.addKeyMapping('keyCode', 'test_shortcut');
+
+      var shortcutNames = this.registry.keyMap_['keyCode'];
+      chai.assert.lengthOf(shortcutNames, 1);
+      chai.assert.equal(shortcutNames[0], 'test_shortcut');
+    });
+    test('Adds a colliding key mapping - opt_allowCollision=true', function() {
+      this.registry.registry_['test_shortcut'] = {'name': 'test_shortcut'};
+      this.registry.keyMap_['keyCode'] = ['test_shortcut_2'];
+
+      this.registry.addKeyMapping('keyCode', 'test_shortcut', true);
+
+      var shortcutNames = this.registry.keyMap_['keyCode'];
+      chai.assert.lengthOf(shortcutNames, 2);
+      chai.assert.equal(shortcutNames[0], 'test_shortcut');
+      chai.assert.equal(shortcutNames[1], 'test_shortcut_2');
+    });
+    test('Adds a colliding key mapping - opt_allowCollision=false', function() {
+      this.registry.registry_['test_shortcut'] = {'name': 'test_shortcut'};
+      this.registry.keyMap_['keyCode'] = ['test_shortcut_2'];
+
+      var registry = this.registry;
+      var shouldThrow = function() {
+        registry.addKeyMapping('keyCode', 'test_shortcut');
+      };
+      chai.assert.throws(
+          shouldThrow, Error,
+          'Shortcut with name "test_shortcut" collides with shortcuts test_shortcut_2');
+    });
+  });
+
+  suite('removeKeyMapping', function() {
+    test('Removes a key mapping', function() {
+      this.registry.registry_['test_shortcut'] = {'name': 'test_shortcut'};
+      this.registry.keyMap_['keyCode'] = ['test_shortcut', 'test_shortcut_2'];
+
+      var isRemoved =
+          this.registry.removeKeyMapping('keyCode', 'test_shortcut');
+
+      var shortcutNames = this.registry.keyMap_['keyCode'];
+      chai.assert.lengthOf(shortcutNames, 1);
```

Comment:
I would probably also check that the only shortcut in list is 'test_shortcut_2'

Reply:
done